chain/stateGenerator.go

```diff
@@ -16,6 +16,17 @@ func spendTransaction(states *db.StateDB, tx *Transaction) error {
 	switch tx.UnsignedTx.Payload.Type {
 	case CoinbaseType:
 		coinbase := pl.(*Coinbase)
+		if !genesis {
+			accSender := states.GetOrNewAccount(common.BytesToUint160(coinbase.Sender))
+			amountSender := accSender.GetBalance()
+			donation, err := DefaultLedger.Store.GetDonation()
+			if err != nil {
+				return err
+			}
+			accSender.SetBalance(amountSender - donation.Amount)
```

Comment:
Same as above, we should probably verify amountSender >= donation.Amount just in case donation calculation has a bug.